docs/samples/Microsoft.ML.AutoML.Samples/Sweepable/SweepableLightGBMBinaryExperiment.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.ML.Data;
+using Microsoft.ML.SearchSpace;
+
+namespace Microsoft.ML.AutoML.Samples
+{
+    public static class SweepableLightGBMBinaryExperiment
+    {
+        class LightGBMOption
+        {
+            [Range(4, 32768, init: 4, logBase: false)]
+            public int NumberOfLeaves { get; set; } = 4;
+
+            [Range(4, 32768, init: 4, logBase: false)]
+            public int NumberOfTrees { get; set; } = 4;
+        }
+
+        public static async Task RunAsync()
+        {
+            // This example shows how to use Sweepable API to run hyper-parameter optimization over
+            // LightGBM trainer with a customized search space.
+
+            // Create a new context for ML.NET operations. It can be used for
+            // exception tracking and logging, as a catalog of available operations
+            // and as the source of randomness. Setting the seed to a fixed number
+            // in this example to make outputs deterministic.
+            var seed = 0;
+            var context = new MLContext(seed);
+
+            // Create a list of training data points and convert it to IDataView.
+            var data = GenerateRandomBinaryClassificationDataPoints(100, seed);
+            var dataView = context.Data.LoadFromEnumerable(data);
+
+            var trainTestSplit = context.Data.TrainTestSplit(dataView);
```

Comment:
It might help to add a comment here saying that the default split is 90/10. You could also add the split ratio as well. In either case, let's just make sure to comment it.